internal/k8s/reconciler/route.go

```diff
@@ -177,15 +186,15 @@ func (r *K8sRoute) Parents() []gwv1alpha2.ParentReference {
 	return nil
 }
 
-func (r *K8sRoute) Validate(ctx context.Context) error {
+func (r *K8sRoute) validate(ctx context.Context) error {
```

Comment:
Did we want to strip this out at this point? I'm all for keeping the method surface area for these fairly slim just so we don't wind up fattening up the route and gateway wrappers in the future..

Reply:
It's on my list, but I'm planning to do it in a separate PR as similar changes will need to happen for the gateway